ledger/participant-state/kvutils/tools/codec-benchmark/src/benchmark/scala/com/daml/lf/benchmark/TypedValueExtractor.scala

```diff
@@ -34,8 +34,17 @@ final class TypedValueExtractor(signatures: PartialFunction[PackageId, PackageSi
       node.getNodeTypeCase match {
         case NodeTypeCase.CREATE =>
           val create = node.getCreate
-          val (packageId, qualifiedName) =
-            validateIdentifier(create.getContractInstance.getTemplateId)
+          val (packageId, qualifiedName) = {
+            import scala.Ordering.Implicits.infixOrderingOps
+            TransactionCoder.decodeVersion(node.getVersion) match {
+              case Right(ver) if (ver >= TransactionVersion.V12) =>
+                validateIdentifier(node.getCreate.getTemplateId)
+              case Right(_) =>
+                validateIdentifier(create.getContractInstance.getTemplateId)
```

Comment:
What does `node.getCreate.getTemplateId` resolve to for LF < 1.12? In other words, why can’t we always use this?

Reply:
`node` is a protobuf message. 
 - in v < 12  the node Create has a `contractInstance` (which contains a versioned value, the template Id and the agreement text.)

Reply:
oh I see, makes sense :+1:

Reply:
I used `node.getCreate` instead of `create` at line 41. Maybe this was confusing you.

Reply:
Sorry for the mess up with the comments. I was not very sure what was the issue you had.

For the reference the `contractInstance` is inline inside in the create node. See #8519 and #8526.

Reply:
I think I was just tired when reviewing it and didn’t review it properly. All makes sense to me now :+1: